### ISSUE TITLE
Run peak prominence / topographic isolation algorithm to display only prominent mountain peaks

### ISSUE THREAD

**Issue by hiddewie** [organicmaps/organicmaps#9190]
**Is your feature request related to a problem? Please describe.**

Currently, mountain peaks are shown from the first zoom level where downloaded country details are shown.

Every mountain peak is shown on the map, irrespective of its elevation.

(see screenshot)

Having fewer but more important mountain peaks on the map ensures the map is less cluttered, and more targeted. Especially together with contour lines, this creates a nice overview visualization of a region for the lower zoom levels.

The main OSM carto visualization has the same problem, see https://github.com/gravitystorm/openstreetmap-carto/issues/3722. However, that data has to be updatable dynamically without reprocessing the entire world when a single mountain peak changes. The OrganicMaps OSM data can be post-processed with algorithms before visualizing it because it is not updated every minute.

**Describe the ideal solution**

Run the prominence / topographic isolation algorithm on the mountain peak data. This produces a number for each peak, how prominent it is in the direct area.

The map can display peaks of a certain prominence depending on the zoom level.

Related documentation and implementations:
- https://osm2pgsql.org/doc/manual.html#strategy-discrete-isolation
- https://github.com/MathiasGroebe/discrete_isolation
- https://wiki.openstreetmap.org/wiki/User:Maxbe/Dominanz_von_Gipfeln (in German)
- C implementation https://github.com/der-stefan/OpenTopoMap/blob/e4467cfc2064afc379b0f8e8360db1740099cca3/mapnik/tools/isolation.c

**Describe alternatives you have considered**

Other maps also use this solution, if the resources are available to recalculate the peak prominence when the data changes. For example OpenTopoMap.

**Additional context**

If this feature is not worth the processing or complexity, feel free to close this issue.

![image](https://github.com/user-attachments/assets/03c41fd0-8421-4eff-adb2-f35e325fc50a)

**Comment by map-per** [comment]
This looks like a cool, but complicated feature.

If I understand it right your proposal is something different than the [OSM prominence tag](https://wiki.openstreetmap.org/wiki/Key:prominence) and the proposed "importance" value is the distance to the closest peak that's higher.

**Comment by hiddewie** [comment]
> If I understand it right your proposal is something different than the [OSM prominence tag](https://wiki.openstreetmap.org/wiki/Key:prominence)

Yes exactly. I don't propose to add this kind of data into OSM, because it would not be easily verifiable. In particular the `prominence` tag has 9000 usages worldwide, while there are almost 1000000 peaks mapped.

Rather, the prominence / importance of a peak is a calculated value. I see it as a heuristic, so the exact value does not matter but it allows ranking the peaks against other peaks in the region. This works well when looking at the result as a human.

The importance threshold which peaks are visualized can be tweaked per application / zoom level.
